### PR TITLE
[cdac][.NET9] Data descriptor changes for GetMethodDescData

### PR DESCRIPTION
**Pull request by lambdageek**
Extracted from https://github.com/dotnet/runtime/pull/106413

The main new thing here is a `PrecodeMachineDescriptor` struct and global: this is used to copy out the `enum` values and masks that show how to manipulate precode stubs to figure out what kind of a stub we're looking at.

Contributes to https://github.com/dotnet/runtime/issues/99302

### CODE REVIEW

**Comment by carlossanlop** [comment]
@jkotas @AaronRobinsonMSFT @davidwrighton @elinor-fung please help reviewing this PR. @lambdageek has requested that I wait for it before the RC1 snap.

File: src/coreclr/vm/loaderallocator.hpp [hunk 1]
```diff
@@ -332,7 +332,7 @@ class LoaderAllocator
     bool                m_fTerminated;
     bool                m_fMarked;
     int                 m_nGCCount;
-    bool                m_IsCollectible;
+    BYTE                m_IsCollectible;
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/loaderallocator.hpp:335]:
```suggestion
    const BYTE          m_IsCollectible;
```

**Review by AaronRobinsonMSFT — COMMENTED** [review]
Do we need to update any markdown?

File: src/coreclr/vm/precode.cpp [hunk 2]
```diff
@@ -662,4 +662,55 @@ BOOL DoesSlotCallPrestub(PCODE pCode)
     return FALSE;
 }
 
+PrecodeMachineDescriptor g_PrecodeMachineDescriptor;
+
+void PrecodeMachineDescriptor::Init()
```
Comment (davidwrighton) [inline src/coreclr/vm/precode.cpp:667]: Why is there an Init routine as well as a set of non-static data member initializers here?

Comment (lambdageek) [reply src/coreclr/vm/precode.cpp:667]: I need to delete the initializers, that was an idea that didn't work out

Comment (davidwrighton) [reply src/coreclr/vm/precode.cpp:667]: Are you doing this before you want this checked in? Given the snap deadline, I think we should probably miss it, and do a backport of the data descriptors when we're ready in reality to declare that .NET 9 might actually be useable with the cDAC and CLRMA. In general, I'm leery of stuff that has a static initializer, and that's what we have here now, and I don't see us fixing this before the snap happens.

Comment (lambdageek) [reply src/coreclr/vm/precode.cpp:667]: Yea, let's miss the snap. I can't fix and validate this in the next couple hours

**Comment by lambdageek** [comment]
@carlossanlop I'm going to close this and do a back port after the snap, it needs more work

**Pull request closed without merging**